src/library-authoring/create-collection/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,131 @@
+      showToast(intl.formatMessage(messages.createCollectionSuccess));
+    }).catch(() => {
+      setIsCreatingCollection(false);
+      showToast(intl.formatMessage(messages.createCollectionError));
+    });
+  }, []);
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.createCollectionModalTitle)}
+      isOpen={isCreateCollectionModalOpen}
+      onClose={handleOnClose}
+      size="xl"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {intl.formatMessage(messages.createCollectionModalTitle)}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+
+      <Formik
+        initialValues={{
+          title: '',
+          description: '',
+        }}
+        validationSchema={
+          Yup.object().shape({
+            title: Yup.string()
+              .required(intl.formatMessage(messages.createCollectionModalNameInvalid)),
+            description: Yup.string(),
+          })
+        }
+        onSubmit={(values) => handleCreate(values)}
+      >
+        {(formikProps) => (
+          <>
+            <ModalDialog.Body className="mw-sm">
+              <Form onSubmit={formikProps.handleSubmit}>
+                <FormikControl
+                  name="title"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createCollectionModalNameLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.title}
+                  placeholder={intl.formatMessage(messages.createCollectionModalNamePlaceholder)}
+                  help=""
+                  className=""
+                  controlClasses="pb-2"
+                />
+                <FormikControl
+                  name="description"
+                  as="textarea"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createCollectionModalDescriptionLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.description}
+                  placeholder={intl.formatMessage(messages.createCollectionModalDescriptionPlaceholder)}
+                  help={intl.formatMessage(messages.createCollectionModalDescriptionDetails)}
+                  className=""
+                  controlClasses="pb-2"
+                  rows="5"
+                />
+              </Form>
+            </ModalDialog.Body>
+            <ModalDialog.Footer>
+              <ActionRow>
+                <ModalDialog.CloseButton variant="tertiary">
+                  {intl.formatMessage(messages.createCollectionModalCancel)}
+                </ModalDialog.CloseButton>
+                <Button
+                  variant="primary"
+                  onClick={formikProps.submitForm}
+                  disabled={isCreatingCollection}
```

Comment:
And you can use the `isSubmitting` here.
Optionally, the `isValid` and `dirty` will only allow the form to be submitted after the user enters some valid data.

```suggestion
                  disabled={formikProps.isSubmitting || !formikProps.isValid || !formikProps.dirty}
```

Reply:
Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1259/commits/636d2312451f3c3745591292de2ffd510eb40294